sugarcrm/client.py

```diff
@@ -0,0 +1,70 @@
+import requests
+import json
+import hashlib
+import sugarcrm
+from sugarcrm.error import CrmAuthenticationError, CrmResponseError
+
+
+class APIClient:
+    def __init__(self, client=requests, logger=sugarcrm.logger):
+        self.client = client
+        self.session_id = None
+        self.rest_url = sugarcrm.rest_url
+        self.logger = logger
+
+    def _api_request(self, method, args):
```

Comment:
I think this is the perfect place to ensure we are logged in before issuing the request. It would as simple as checking it there's a `self.session_id` yet, I think. If not, we log in first.

Reply:
Done.